src/service/resources-impl.js

```diff
@@ -600,6 +601,22 @@ export class Resources {
     });
   }
 
+  collapseElement(element) {
+    const box = this.viewport_.getLayoutRect(element);
+    const resource = Resource.forElement(element);
+    if (box.width != 0 && box.height != 0) {
+      this.setRelayoutTop_(box.top);
```

Comment:
I looked into this. The change heights requests will be executed as [part](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L706-L811) of [`#mutateWork_`](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L674). The code paths that [depend on `relayoutTop`](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L867-L868) are in [`#discoverWork`](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L857-L992), which is executed [after](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1411-L1414) the request height changes are executed.

So, we'll only update the `relayoutTop` value (maybe, if the new value is smaller) in this pass, not schedule another pass with this `box.top`.